.eslintrc.json

```diff
@@ -1,24 +1,27 @@
 {
-  "env": {
-    "node": true,
-    "commonjs": true,
-    "es2017": true,
-    "mocha": true
-  },
-  "ignorePatterns": ["**/*.mjs", "**/*.js"],
+  "ignorePatterns": ["build", "coverage"],
   "extends": [
     "eslint:recommended",
     "plugin:@typescript-eslint/recommended-type-checked",
     "plugin:@typescript-eslint/stylistic-type-checked",
+    "plugin:import/recommended",
+    "plugin:import/typescript",
     "plugin:prettier/recommended"
   ],
-  "plugins": ["@typescript-eslint", "import"],
+  "settings": {
+    "import/resolver": {
+      "typescript": true
+    }
+  },
   "parserOptions": {
-    "ecmaVersion": 9,
     "project": true
   },
-  "rules": {
-    "import/no-default-export": "error",
-    "import/extensions": ["error", "always"]
-  }
+  "overrides": [
+    {
+      "files": ["lib", "test"],
+      "rules": {
+        "import/no-default-export": "error"
+      }
+    }
```

Comment:
Since we might still use default exports in stuff like config files, I've limited this rule to the source and tests.